### PR TITLE
Add offset to find published comments, added function to parse additional data and updated the comment events

### PR DESCRIPTION
**Pull request by martinlagler**
…tional data and updated the comment events

| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

Added offset to get comments function.
Changed post events so they are dispatched after the flush.
Added endpoint to count the comments.
Added possibility to parse additional data to the comments twig file.

### CODE REVIEW

File: Controller/WebsiteCommentController.php [hunk 1]
```diff
@@ -121,12 +121,14 @@ public function cgetCommentsAction(string $threadId, Request $request): Response
...
         $pageSize = $request->get('pageSize') ?? 20;
+        $pageOffset = $request->get('offset') ?? 0;
 
         $comments = $this->commentManager->findPublishedComments(
             $type,
             $entityId,
             $page ?: 1,
-            $page ? $pageSize : null
+            $page ? $pageSize : null,
+            $pageOffset
```
Comment (alexander-schranz) [inline Controller/WebsiteCommentController.php]: Have an `offset` and a `page` is very strange, this should not be the case.

Comment (wachterjohannes) [inline Controller/WebsiteCommentController.php]: would also use here only page/pageSize

File: Controller/WebsiteCommentController.php [hunk 2]
```diff
@@ -150,20 +152,57 @@ public function cgetCommentsAction(string $threadId, Request $request): Response
...
             $this->twig->render(
                 $this->getTemplate($type, 'comments'),
+                $contentData
+            )
+        );
+
+        return $response;
+    }
+
+    protected function getAdditionalContentData(Request $request): array
```
Comment (wachterjohannes) [inline Controller/WebsiteCommentController.php:198]: missing @return phpdoc

File: Entity/CommentRepository.php [hunk 3]
```diff
@@ -54,7 +59,9 @@ public function findPublishedComments(string $type, string $entityId, int $page
 
         if ($pageSize) {
             $query->setMaxResults($pageSize);
-            $query->setFirstResult(($page - 1) * $pageSize);
+            $query->setFirstResult(($page - 1) * $pageSize + $offset);
+        } elseif ($offset) {
```
Comment (wachterjohannes) [inline Entity/CommentRepository.php]: would remove offset also here - as it colides with pageSize

File: Manager/CommentManager.php [hunk 4]
```diff
@@ -180,6 +193,6 @@ private function deleteThread(ThreadInterface $thread): void
 
         $this->threadRepository->delete($thread);
 
-        $this->dispatcher->dispatch(new ThreadEvent($thread), Events::THREAD_POST_DELETE_EVENT);
+        $this->commentEventCollector->collect(new ThreadEvent($thread), Events::THREAD_POST_DELETE_EVENT);
```
Comment (wachterjohannes) [inline Manager/CommentManager.php:191]: would also do that for publish/unpublish

**Review by wachterjohannes — APPROVED** [review]

File: Controller/WebsiteCommentController.php [hunk 5]
```diff
@@ -118,17 +118,31 @@ public function cgetCommentsAction(string $threadId, Request $request): Response
     {
         list($type, $entityId) = $this->getThreadIdParts($threadId);
 
-        $page = $request->get('page');
+        $limit = $request->query->getInt('limit') ?? 10;
+        $offset = $request->query->getInt('offset') ?? 0;
+
+        $pageSize = $request->get('pageSize') ?? 10;
+        if ($pageSize) {
```
Comment (alexander-schranz) [inline Controller/WebsiteCommentController.php]: This is always true.

```suggestion
        $pageSize = $request->get('pageSize');
        if ($pageSize) {
```

File: Controller/WebsiteCommentController.php [hunk 6]
```diff
@@ -118,17 +118,31 @@ public function cgetCommentsAction(string $threadId, Request $request): Response
...
+            @\trigger_deprecation('sulu/comment-bundle', '2.x', 'The usage of the "pageSize" parameter is deprecated.
+        Please use "limit" and "offset instead.');
+            $limit = $pageSize;
+        }
+
+        $page = $request->get('page') ?? null;
+        if ($page) {
```
Comment (alexander-schranz) [inline Controller/WebsiteCommentController.php]: `?? null` is unnecessary `??` says `if value == null` and if it is already null we don't need to set it to null.

```suggestion
        $page = $request->get('page');
        if ($page) {
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Review by wachterjohannes — APPROVED** [review]

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**